### DISCUSSION TITLE
`stage(start = variable)` works by itself without specifying `after_stat`/`after_scale` 😄🥳

### DISCUSSION THREAD

**Discussion by yjunechoe** [ggplot2-extenders/ggplot-extension-club#44] [General]
For the last part of my delayed aesthetics blog post series (which I've been putting off for like 2 years now 🙃), I've always been wanting to use this pedagogical pivot of "every aesthetic that's ordinarily specified is conceptually equivalent to `stage()`-ing it at the start and not doing anything further with it at later stages". 

```r
ggplot(mtcars, aes(cyl)) +
  stat_count()

ggplot(mtcars, aes(stage(start = cyl)) +
  stat_count()
```

And I kind of lost hope of being able to do this after https://github.com/tidyverse/ggplot2/pull/4890 was closed and https://github.com/tidyverse/ggplot2/issues/4873 was merged. But it looks like this _just works_ now - not sure when this happened, but presumably thanks to some recent changes to how `stage()` is handled? (@teunbrand pls let this behavior stay 🥺)

This means I can introduce `stage()` more gracefully/gradually and makes strategies like the `stage(after_scale)` trick only 1-step removed :D

```r
library(ggplot2)

p1 <- mtcars |>
  transform(cyl = as.factor(cyl)) |>
  ggplot(aes(cyl)) +
  stat_count(
    aes(fill = stage(cyl))
  )

p2 <- mtcars |>
  transform(cyl = as.factor(cyl)) |>
  ggplot(aes(cyl)) +
  stat_count(
    aes(fill = stage(cyl, after_scale = colorspace::darken(fill, .5)))
  )

patchwork::wrap_plots(p1, p2)
```

![image](https://github.com/user-attachments/assets/e140a343-2065-46dc-9615-5445b56c4641)

**Comment by teunbrand** [comment]
> not sure when this happened

I also don't know, but I'm reasonably sure it must be some unconsidered side effect of fixing some other issue 😅 
For stuff that you think might work 'accidentally', just send a PR to ggplot2 with a unit test for the behaviour you want to keep.
When there are no tests for a thing, people might just go and break the thing by accident, but otherwise they have to intentionally break it.
If the PR gets rejected you are probably in ['off-label'](https://www.tidyverse.org/blog/2021/06/off-label-uses-in-ggplot2/) territory.

**Reply by yjunechoe** [reply]
> For stuff that you think might work 'accidentally', just send a PR to ggplot2 with a unit test for the behaviour you want to keep.

That makes sense! Thanks for the pointer

After poking around a bit, this seems to have happened at https://github.com/tidyverse/ggplot2/pull/5409. There are no existing tests for the behavior of `stage(variable)`/`stage(start = variable)` - if I were to write one, I assume that would goes under `test-aes-calculated.R`?

**Reply by teunbrand** [reply]
Good sleuthing, I now recall making the comment on the PR 😁 

> I assume that would goes under test-aes-calculated.R?

Seems like a good place for it :)
